src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -206,13 +207,23 @@ def _get_user_id(self, admin_path, request=None):
         return response['data']['user_id']
 
     @RestClient.api_get('/{admin_path}/metadata/user', resp_structure='[+]')
-    def _is_system_user(self, admin_path, request=None):
+    def _user_exists(self, admin_path, request=None):
         # pylint: disable=unused-argument
         response = request()
         return self.userid in response
 
+    def user_exists(self):
+        return self._user_exists(self.admin_path)
+
+    @RestClient.api_get('/{admin_path}/metadata/user?key={userid}',
+                        resp_structure='data > system')
+    def _is_system_user(self, admin_path, userid, request=None):
+        # pylint: disable=unused-argument
+        response = request()
+        return strtobool(response['data']['system'])
+
     def is_system_user(self):
-        return self._is_system_user(self.admin_path)
```

Comment:
Is anyone providing something different than `self.admin_path`?

Reply:
See my previous [comment](https://github.com/ceph/ceph/pull/22669#discussion_r198736575).